lib/datatables_crud/datatables/active_record_datatable.rb

```diff
@@ -30,27 +30,49 @@ def sort_options
       end
 
       def count_options
-        (@options || {}).reject { |k, v| %w(limit offset).include? k.to_s }
+        (@options || {}).reject { |k, v| %w(limit offset order).include? k.to_s }
       end
 
       def count
         if params[:search].try(:[], :value).present? and search_columns.present?
-          @count ||= prepared_clazz.where(search_columns.map { |v| "#{v} like :search" }.join(' OR '), search: "%#{params[:search][:value]}%").count(count_options)
+          @count ||= count_of( prepared_clazz.where(search_columns.map { |v| "#{v} like :search" }.join(' OR '), search: "%#{params[:search][:value]}%") )
         else
           total_count
         end
       end
 
       def total_count
-        @total_count ||= prepared_clazz.count(count_options)
+        @total_count ||= count_of(prepared_clazz)
+      end
+
+      def count_of(objects)
+        objects.where(count_options[:conditions]).count
       end
 
       def records
+        @records_exist = @records.present?
```

Comment:
Why is @records_exist assigned before we load the records?

Reply:
I added this @records_exist to avoid running the options queries again.
if @records already exists so this line for example
`@records ||= prepared_clazz.all`
will not change the @records so we don't need to run the options queries again (conditions, limit ..etc)
That's why I am checking the @record_exists before applying the options.

Reply:
`return unless @records` will do exactly that. I think @records_exist is redundant. Please remove it.

Reply:
I see your point, but I think it's ok. Either rails or mysql will handle this efficiently.